pkg/rc/replication_controller.go

```diff
@@ -744,10 +744,10 @@ func (rc *replicationController) canNodeTransfer(rcFields fields.RC, current typ
 	switch {
 	case consul.IsAlreadyLocked(err):
 		rc.logger.Infoln("skipping node transfer; rc mutation lock is already held")
-		return false, nil, nil
+		return false, unlocker, nil
 	case err != nil:
 		rc.logger.WithError(err).Errorln("could not acquire rc mutatation lock for node transfer")
-		return false, nil, err
+		return false, unlocker, err
```

Comment:
My suspicion is that this is the case we are hitting that leaves us with orphaned locks, but I am not confident that such is the case. The alternative is we have multiple workers all creating locks with the same RC ID.